core-bundle/src/Twig/Extension/ContaoExtension.php

```diff
@@ -280,6 +281,11 @@ public function getFilters(): array
                 [SanitizerRuntime::class, 'sanitizeHtml'],
                 ['is_safe' => ['html']]
             ),
+            new TwigFilter(
+                'encode_email',
+                [StringRuntime::class, 'encodeEmail'],
+                ['is_safe' => ['html']]
```

Comment:
This would bypass the escaping of the rest of the text, so `{{ '<script>…'|encode_email }}` would no longer be safe 🙈

We probably need to use a `ChunkedText` object here.

Reply:
Thinking about this, we maybe need to create an escaper filter for email, because otherwise having an escaper filter afterwards will always be a problem. OR the filter needs to enable preescaping. But I'm not sure that either is the best/correct solution. 🤔

Reply:
Since the filter only encodes the email address and does not add an `<a href="mailto:…"` link, can we not omit this line entirely or use `preserves_safety` instead?

Reply:
Yes, `preserves_safety` should work for now I think (as we use `double_encode: false`)

Reply:
Should probably document that as comment as well :)

Reply:
We are only using `double_encode: false` for certain templates. And what about other contexts?

Reply:
As discussed in the Contao call, we want to use `['preserves_safety' => ['html']]`, so the filter works in our default context. In other contexts, you have to add `|raw` to make the filter work, and if it should be required at some point in the future, we could come up with a more sophisticated solution.

Reply:
It might work if we say it to `['contao_html', 'html']`. The evaluation of that happens independently from our escaper node visitor.

Reply:
Changed in b346c95bdca1d14c14b965e5d2cd63b4cc0224c4.